builder/amazon/common/step_pre_validate.go

```diff
@@ -76,13 +78,34 @@ func (s *StepPreValidate) Run(ctx context.Context, state multistep.StateBag) mul
 		ui.Say("Force Deregister flag found, skipping prevalidating AMI Name")
 		return multistep.ActionContinue
 	}
+
 	if s.AMISkipBuildRegion {
 		ui.Say("skip_build_region was set; not prevalidating AMI name")
 		return multistep.ActionContinue
 	}
 
 	ec2conn := state.Get("ec2").(*ec2.EC2)
```

Comment:
if you do end up creating unit tests for this using the mock, you'll need to change this to cast to a *ec2iface.EC2API instead.